src/rules/infinite-reactive-loop.ts

```diff
@@ -0,0 +1,439 @@
+  )
+}
+
+/**
+ * Return true if `node` is inside of `then` or `catch`.
+ */
+function isPromiseThenOrCatchBody(node: TSESTree.Node): boolean {
+  if (!getDeclarationBody(node)) return false
+  const { parent } = node
+  if (
+    parent?.type !== "CallExpression" ||
+    parent?.callee?.type !== "MemberExpression"
+  ) {
+    return false
+  }
+  const { property } = parent.callee
+  if (property?.type !== "Identifier") return false
+  return ["then", "catch"].includes(property.name)
+}
+
+/**
+ * Get all tracked reactive variables.
+ */
+function getTrackedVariableNodes(
+  context: RuleContext,
+  ast: AST.SvelteReactiveStatement,
+) {
+  const reactiveVariableNodes: TSESTree.Identifier[] = []
+  traverseNodes(ast.body, {
+    enterNode(node) {
+      if (isReactiveVariableNode(context, node)) {
+        reactiveVariableNodes.push(node)
+      }
+    },
+    leaveNode() {
+      /* noop */
+    },
+  })
+  return reactiveVariableNodes
+}
+
+/**  */
+function getDeclarationBody(
+  node: TSESTree.Node,
+  functionName?: string,
+): TSESTree.BlockStatement | TSESTree.Expression | null {
+  if (
+    node.type === "VariableDeclarator" &&
+    node.id.type === "Identifier" &&
+    (!functionName || node.id.name === functionName)
+  ) {
+    if (
+      node.init?.type === "ArrowFunctionExpression" ||
+      node.init?.type === "FunctionExpression"
+    ) {
+      return node.init.body
+    }
+  } else if (
+    node.type === "FunctionDeclaration" &&
+    node.id?.type === "Identifier" &&
+    (!functionName || node.id?.name === functionName)
+  ) {
+    return node.body
+  } else if (!functionName && node.type === "ArrowFunctionExpression") {
+    return node.body
+  }
+  return null
+}
+
+/**  */
+function getFunctionDeclarationNode(
+  functionCall: TSESTree.Identifier,
+): TSESTree.BlockStatement | TSESTree.Expression | null {
+  let parent: AST.SvelteScriptElement | TSESTree.Node | undefined = functionCall
+  let declaration: TSESTree.BlockStatement | TSESTree.Expression | null = null
+
+  while (parent) {
+    if (declaration) return declaration
+    parent = parent.parent as
+      | AST.SvelteScriptElement
+      | TSESTree.Node
+      | undefined
+    if (parent && parent.type === "BlockStatement") {
+      traverseNodes(parent, {
+        // eslint-disable-next-line no-loop-func -- ignore
+        enterNode(node) {
+          if (!declaration) {
+            declaration = getDeclarationBody(node, functionCall.name)
+          }
+        },
+        leaveNode() {
+          /* noop */
+        },
+      })
+    } else if (parent && parent.type === "SvelteScriptElement") {
+      for (const node of parent.body) {
+        if (declaration) break
+        if (node.type === "VariableDeclaration") {
+          for (const child of node.declarations) {
+            declaration = getDeclarationBody(child, functionCall.name)
+            if (declaration) break
+          }
+        }
+      }
+    }
+  }
+
+  return declaration
+}
```

Comment:
Note the addition of the `context` argument in the suggestion.

```suggestion
function getFunctionDeclarationNode(
  context: RuleContext,
  functionCall: TSESTree.Identifier,
): TSESTree.BlockStatement | TSESTree.Expression | null {
  const variable = findVariable(context, functionCall)
  if (!variable) {
    return null
  }
  for (const def of variable.defs) {
    if (def.type === "FunctionName") {
      if (def.node.type === "FunctionDeclaration") {
        return def.node.body
      }
    }
    if (def.type === "Variable") {
      if (
        def.node.init &&
        (def.node.init.type === "FunctionExpression" ||
          def.node.init.type === "ArrowFunctionExpression")
      ) {
        return def.node.init.body
      }
    }
  }
  return null
}
```